src/snowflake/snowpark/_internal/open_telemetry.py

```diff
@@ -56,6 +61,59 @@ def open_telemetry_context_manager(func, dataframe):
         yield
 
 
+@contextmanager
+def open_telemetry_udf_context_manager(func, parameters):
+    # trace when required package is installed
+    if open_telemetry_found:
+        class_name = func.__qualname__
+        name = func.__name__
+        tracer = trace.get_tracer(extract_tracer_name(class_name))
+        with tracer.start_as_current_span(name) as cur_span:
+            try:
+                # first try to get func if it is udf, then try to get handler if it is udtf/udaf, if still None, means it is
+                # loading from file
+                udf_func = (
+                    parameters.get("func")
+                    if parameters.get("func")
+                    else parameters.get("handler")
+                )
+                # if udf_func is not None, meaning it is a udf function or udf handler class, get handler_name from it, otherwise find
+                # function name or handler name from parameter
+                handler_name = (
+                    udf_func.__name__
+                    if udf_func
+                    else (
+                        parameters.get("func_name")
+                        if parameters.get("func_name")
+                        else parameters.get("handler_name")
+                    )
+                )
+                if cur_span.is_recording():
+                    # store execution location in span
+                    filename, lineno = context_manager_code_location(
+                        inspect.stack(), func
+                    )
+                    cur_span.set_attribute("code.filepath", f"{filename}")
+                    cur_span.set_attribute("code.lineno", lineno)
+                    cur_span.set_attribute(
+                        "snow.executable.name", parameters.get("name")
+                    )
+                    cur_span.set_attribute("snow.executable.handler", handler_name)
+                    cur_span.set_attribute(
+                        "snow.executable.filepath", parameters.get("file_path")
+                    )
+            except Exception as e:
+                logger.warning(f"Error when acquiring span attributes. {e}")
+            finally:
+                try:
+                    yield
+                except Exception as e:
+                    cur_span.set_status(Status(StatusCode.ERROR, str(e)))
```

Comment:
I think you can use `trace.Status` instead and avoid import as well. It is clearer which status you are referring to.